src/runtime/methodbinder.cs

```diff
@@ -201,6 +273,17 @@ internal static int ArgPrecedence(Type t)
                 return 3000;
             }
 
+            // Due to array type must be a object, "IsArray" should check first.
+            if (t.IsArray)
```

Comment:
Was this code never called before you moved it up? o-O

Reply:
Yep, it will return by case `case TypeCode.Object:`